### ISSUE TITLE
'make docs' fails when building PDF

### ISSUE THREAD

**Issue by duncan-roe** [doxygen/doxygen#8935]
The HTML builds fine, then I get these errors
`[100%] Generating Latex and HTML documentation.`
`[100%] Built target run_doxygen
`[100%] Generating Doxygen Manual PDF.
`This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021 on Slackware) (preloaded format=pdflatex)`
` \write18 enabled.`
`entering extended mode`
`(./doxygen_manual.tex`
`LaTeX2e <2020-10-01> patch level 4`
`L3 programming layer <2021-02-18>`
`make[3]: *** [doc/CMakeFiles/doxygen_pdf.dir/build.make:75: doc/CMakeFiles/doxygen_pdf] Error 1`
`make[2]: *** [CMakeFiles/Makefile2:661: doc/CMakeFiles/doxygen_pdf.dir/all] Error 2`
`make[1]: *** [CMakeFiles/Makefile2:642: doc/CMakeFiles/docs.dir/rule] Error 2`
`make: *** [Makefile:369: docs] Error 2`

This is from commit d2276288

**Comment by albert-github** [comment]
Unfortunately doesn't give a lot of information, we would need at least the `doxygen_manual.log` file from the `latex` sub directory in the build directory.
- What happens when you do (might be some corrupted files):
   ```
   rm -rf html build
   make docs
   ```

**Comment by duncan-roe** [comment]
Log shows
```
! LaTeX Error: File `xpatch.sty' not found.
```
I tried:
```
$ locate xpatch
/usr/share/texmf-dist/tex/latex/regexpatch
/usr/share/texmf-dist/tex/latex/regexpatch/regexpatch.sty
/usr/share/texmf-dist/tex/latex/skeyval/skeyval-ltxpatch.sty
$
```
[doxygen_manual.log](https://github.com/doxygen/doxygen/files/7652396/doxygen_manual.log)

**Comment by duncan-roe** [comment]
Tried:
```
rm -rf html latex
make docs
```
Same error as before

**Comment by duncan-roe** [comment]
manual.pdf builds with this patch
```
--- a/doc/manual.sty
+++ b/doc/manual.sty
@@ -45,6 +45,6 @@
   \markright{\thesection\ #1}%
 }
 %
-\usepackage{xpatch}
+\usepackage{regexpatch}
 \xpatchcmd{\part}{plain}{empty}{}{}
 
```

**Comment by albert-github** [comment]
From https://ctan.org/pkg/xpatch I have not the idea that this package is not part of the Tex 2021 distribution.
The https://ctan.org/pkg/regexpatch package would be an alternative as it states: "The package generalises the macro patching commands provided by P. Lehmann’s etoolbox. The difference between this package and its sibling xpatch is that this package sports a very powerful \regexpatchcmd based on the l3regex module of the LaTeX3 experimental packages. "

So I'm a bit reluctant tho use it.

As I think that `xpatch.sty` should be provided by Tex 2021 either
- the Slackware distribution is incomplete
- the xpatch package has not been installed when installing the Tex version 
   - user's choice
   - problem in the Slackware package (i.e. again the first mentioned point).

**Comment by duncan-roe** [comment]
There is no mention of `xpatch` in the code. But there are 2 occurrences of `regexpatch`
```
$ sfl regexpatch
src/translator_sk.h:45:             "\\usepackage{regexpatch}\n"
src/translator_cz.h:102:             "\\usepackage{regexpatch}\n"
```
So I think the patch is sound

**Comment by duncan-roe** [comment]
> As I think that `xpatch.sty` should be provided by Tex 2021 either
> 
>     * the Slackware distribution is incomplete
> 
>     * the xpatch package has not been installed when installing the Tex version
>       
>       * user's choice
>       * problem in the Slackware package (i.e. again the first mentioned point).

This is Slackware Current (a.k.a. Slackware 15.0 Beta).  It builds texlive from 2 tarballs:
```
texlive-base-210418.tar.xz
texlive-20210324-source.tar.xz
```
`xpatch.sty` is not in either of these. Searching for `xpatch` finds:
```
-rw-r--r-- root/root     26367 2021-03-24 09:29 texmf-dist/tex/latex/regexpatch/regexpatch.sty
-rw-r--r-- root/root     14706 2013-05-17 08:30 texmf-dist/tex/latex/skeyval/skeyval-ltxpatch.sty
```
in `texlive-base-210418.tar.xz` (and no matches in `texlive-20210324-source.tar.xz`).

**Comment by albert-github** [comment]
I don't know how the Tex packages are build, but the both distributions I have both, MikTeX 2.9, and for Cygwin (TeX Live 2021/Cygwin), both up to date,  have the `xpatch` package in their distribution. Of course Tex / LaTeX is a "kernel" with a number of packages on top of it and on Cygwin I installed the complete set of texlive packages available (and I don't know from which sub package the xpatch came). On MikTeX the system automatically installs missing packages. 

It might be that https://tex.stackexchange.com/questions/158700/latex-cant-find-sty-files-altough-packages-are-installed-texlive-ubuntu-12 gives some useful suggestions for you although it is for Ubuntu. 

Diving a bit into the doxygen's usage of `xpatch`. Doxygen uses it only for `\xpatchcmd{\part}{plain}{empty}{}{}` in the doxygen manual. The result is that on the "part" pages (i.e. "User Manual", "Reference Manual", "Developers Manual" and "Appendices") there is at the bottom no line like:
```
Generated by Doxygen 1.9.3
```
Can we use:
- `\patchcmd{\part}{plain}{empty}{}{}`, a small test shows that this looks like to work, though in the documentation of the `\xpatchcmd` theer are mentioned some general limitations of the `\patchcmd`.
- `\usepackage{regexpatch}`, this looks like to give some good results too. When looking in the doxygen source code the `regexpatch` package is used in the translators for the Czech and Slovak languages already.


Even though I still think it is an installation issue I tend a bit more to the usage of the `regexpatch` package as replacement for the `xpatch` package`.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8945

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by duncan-roe** [comment]
@albert-github you only pushed the patch I originally suggested - I thought you had something fancier in mind

**Comment by albert-github** [comment]
@duncan-roe 
At first I was thinking 'can we get remove the xpatch package by using some "simple" command' but after seeing the usage of the `regexpatch` for the mentioned languages I thought it would be better to use the `refexpatch` package.

**Comment by duncan-roe** [comment]
@albert-github Can anything be done about the "part" pages then? Not only do they miss `Generated by Doxygen 1.9.3`, a blank page follows instead.

**Comment by duncan-roe** [comment]
I tried this patch
```
$ diff ../doxygen/doc/manual.sty doc/manual.sty
49c49
< \xpatchcmd{\part}{plain}{empty}{}{}
---
> \regexpatchcmd{\part}{plain}{empty}{}{}
```
but I could see no difference in `doxygen_manual.pdf`. And there was little difference in `doxygen_manual.log`
```
$ diff ../doxygen/build/latex/doxygen_manual.log build/latex/doxygen_manual.log
1c1
< This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021 on Slackware) (preloaded format=pdflatex 2021.10.19)  4 DEC 2021 09:30
---
> This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2021 on Slackware) (preloaded format=pdflatex 2021.10.19)  8 DEC 2021 19:03
5102c5102
<  933944 words of memory out of 5000000
---
>  933921 words of memory out of 5000000
5122c5122
< Output written on doxygen_manual.pdf (325 pages, 1475713 bytes).
---
> Output written on doxygen_manual.pdf (325 pages, 1475726 bytes).
```

**Comment by albert-github** [comment]
Regarding https://github.com/doxygen/doxygen/issues/8935#issuecomment-988300173:
"missing" Generated by and extra blank page, this all has to do with the structure of a book (the doxygen manual is a book (`\documentclass{book}`) and they always are on an odd page (i.e. the right hand side) and the chapers also always start on an odd page, hence the empty page.
The "Part" pages are a division and can be seen as a cover and have have no Generated by.
So this is all by design and nothing further has to be done (in my opinion).

**Comment by duncan-roe** [comment]
OK Great!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by duncan-roe** [comment]
Confirmed fixed at 1.9.3
